### PR TITLE
[ABI Break][SYCL] Remove deprecated is_host() for SYCL objects and related handler ctor param

### PR DESCRIPTION
**Pull request by KseniyaTikhomirova**
Host device support is deprecated long time ago. 
Removes:
1) is_host() API deprecated long time ago, SYCL2020 deprecations group.
2) remove isHost ctor param and corresponding field from handler class. Handler instance is always created by RT. User deals with handler instance by creating kernel lambda which accepts handler as parameter and then calls methods of handler.

### CODE REVIEW

**Review by sarnex — APPROVED** [review]
esimd lgtm

File: sycl/include/sycl/handler.hpp [hunk 1]
```diff
@@ -464,9 +464,8 @@ class __SYCL_EXPORT handler {
   /// Constructs SYCL handler from queue.
   ///
   /// \param Queue is a SYCL queue.
-  /// \param IsHost indicates if this handler is created for SYCL host device.
   /// TODO: Unused. Remove with ABI break.
-  handler(std::shared_ptr<detail::queue_impl> Queue, bool /*Unused*/);
+  handler(std::shared_ptr<detail::queue_impl> Queue);
```
Comment (KseniyaTikhomirova) [inline sycl/include/sycl/handler.hpp:468]: would be great to remove these methods at all

File: sycl/include/sycl/info/info_desc.hpp [hunk 2]
```diff
@@ -58,7 +58,6 @@ enum class device_type : pi_uint64 {
   // TODO: figure out if we need all the below in PI
   custom = PI_DEVICE_TYPE_CUSTOM,
   automatic,
-  host,
```
Comment (steffenlarsen) [inline sycl/include/sycl/info/info_desc.hpp:61]: The [SYCL specification](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#appendix.device.descriptors) still has this, even if we don't intend to use it.

**Review by steffenlarsen — COMMENTED** [review]
Overall I think it looks great, but we need to add back `device_type::host` as it is part of the spec. We don't have to use it anywhere however.

File: sycl/include/sycl/info/info_desc.hpp (hunk 2 above)
Comment (KseniyaTikhomirova) [reply sycl/include/sycl/info/info_desc.hpp]: "host" device_type is brought back https://github.com/intel/llvm/pull/14258/commits/09f3a1cb082665b41cfaabc349c38f4183ca8cf3

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Pull request merged**